TLM/TLM/Custom/AI/CustomTrainAI.cs

```diff
@@ -715,168 +1014,245 @@ public class CustomTrainAI : TrainAI { // TODO inherit from VehicleAI (in order
+                                             ref bool overlap,
+                                             Vector3 min,
+                                             Vector3 max) {
+            Log._DebugOnlyError("CustomTrainAI.CheckOverlap (2) called.");
             return 0;
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static void InitializePath(ushort vehicleID, ref Vehicle vehicleData) {
-            Log.Error("CustomTrainAI.InitializePath called");
+        [UsedImplicitly]
+        private static void InitializePath(ushort vehicleId, ref Vehicle vehicleData) {
+            Log._DebugOnlyError("CustomTrainAI.InitializePath called");
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static void UpdatePathTargetPositions(TrainAI trainAI, ushort vehicleID, ref Vehicle vehicleData, Vector3 refPos1, Vector3 refPos2, ushort leaderID, ref Vehicle leaderData, ref int index, int max1, int max2, float minSqrDistanceA, float minSqrDistanceB) {
-            Log.Error($"CustomTrainAI.InvokeUpdatePathTargetPositions called! trainAI={trainAI}");
+        [UsedImplicitly]
+        public static void UpdatePathTargetPositions(TrainAI trainAi,
+                                                     ushort vehicleId,
+                                                     ref Vehicle vehicleData,
+                                                     Vector3 refPos1,
+                                                     Vector3 refPos2,
+                                                     ushort leaderId,
+                                                     ref Vehicle leaderData,
+                                                     ref int index,
+                                                     int max1,
+                                                     int max2,
+                                                     float minSqrDistanceA,
+                                                     float minSqrDistanceB) {
+            Log._DebugOnlyError($"CustomTrainAI.InvokeUpdatePathTargetPositions called! trainAI={trainAi}");
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static void Reverse(ushort leaderID, ref Vehicle leaderData) {
-            Log.Error("CustomTrainAI.Reverse called");
+        [UsedImplicitly]
+        private static void Reverse(ushort leaderId, ref Vehicle leaderData) {
+            Log._DebugOnlyError("CustomTrainAI.Reverse called");
         }
 
         [RedirectReverse]
         [MethodImpl(MethodImplOptions.NoInlining)]
-        private static float GetMaxSpeed(ushort leaderID, ref Vehicle leaderData) {
-            Log.Error("CustomTrainAI.GetMaxSpeed called");
+        [UsedImplicitly]
+        private static float GetMaxSpeed(ushort leaderId, ref Vehicle leaderData) {
+            Log._DebugOnlyError("CustomTrainAI.GetMaxSpeed called");
```

Comment:
need to call Error, not _DebugOnlyError

Reply:
in Release the function is not eliminated, the Log call is eliminated